<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>StudentWindow</class>
 <widget class="QMainWindow" name="StudentWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>900</width>
    <height>700</height>
   </rect>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QStackedWidget" name="stackedWidget">
      
      <!-- Page 0: Login -->
      <widget class="QWidget" name="loginPage">
       <layout class="QVBoxLayout" name="loginLayout">
        <item alignment="Qt::AlignHCenter">
         <widget class="QLabel" name="loginTitle">
          <property name="text"><string>Student Login</string></property>
         </widget>
        </item>
        <item>
         <spacer name="spacer1">
          <property name="orientation"><enum>Qt::Vertical</enum></property>
         </spacer>
        </item>
        <item>
         <layout class="QFormLayout" name="formLayout">
          <item row="0" column="0">
           <widget class="QLabel" name="studentIdLabel"><property name="text"><string>Student ID:</string></property></widget>
          </item>
          <item row="0" column="1">
           <widget class="QLineEdit" name="studentIdEdit"/>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QPushButton" name="loginButton">
          <property name="text"><string>Login</string></property>
          <property name="minimumHeight"><number>50</number></property>
         </widget>
        </item>
        <item>
         <spacer name="spacer2">
          <property name="orientation"><enum>Qt::Vertical</enum></property>
         </spacer>
        </item>
       </layout>
      </widget>
      
      <!-- Page 1: Instructions -->
      <widget class="QWidget" name="instructionsPage">
       <layout class="QVBoxLayout" name="instrLayout">
        <item alignment="Qt::AlignHCenter">
         <widget class="QLabel" name="instructionsTitle">
          <property name="text"><string>Instructions</string></property>
         </widget>
        </item>
        <item>
         <widget class="QFrame" name="timeLimitFrame">
          <property name="styleSheet"><string>background-color: #e3f2fd; border: 2px solid #1976d2; border-radius: 10px; padding: 15px;</string></property>
          <layout class="QHBoxLayout" name="timeLayout">
           <item>
            <widget class="QLabel" name="timeLimitLabel">
             <property name="text"><string>Time Limit: 30 seconds</string></property>
             <property name="styleSheet"><string>font-size: 18px; color: #1565c0; font-weight: bold; background: transparent; border: none;</string></property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QFrame" name="passingFrame">
          <property name="styleSheet"><string>background-color: #e8f5e9; border: 2px solid #388e3c; border-radius: 10px; padding: 15px;</string></property>
          <layout class="QHBoxLayout" name="passingLayout">
           <item>
            <widget class="QLabel" name="passingLabel">
             <property name="text"><string>Passing Score: 50%</string></property>
             <property name="styleSheet"><string>font-size: 18px; color: #2e7d32; font-weight: bold; background: transparent; border: none;</string></property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QFrame" name="randomFrame">
          <property name="styleSheet"><string>background-color: #fff3e0; border: 2px solid #f57c00; border-radius: 10px; padding: 15px;</string></property>
          <layout class="QHBoxLayout" name="randomLayout">
           <item>
            <widget class="QLabel" name="randomLabel">
             <property name="text"><string>Questions are randomized</string></property>
             <property name="styleSheet"><string>font-size: 18px; color: #e65100; font-weight: bold; background: transparent; border: none;</string></property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <spacer name="spacer3">
          <property name="orientation"><enum>Qt::Vertical</enum></property>
         </spacer>
        </item>
        <item alignment="Qt::AlignHCenter">
         <widget class="QLabel" name="goodLuckLabel">
          <property name="text"><string>Good Luck!</string></property>
          <property name="styleSheet"><string>font-size: 24px; color: #27ae60; font-weight: bold;</string></property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="startExamButton">
          <property name="text"><string>Start Exam</string></property>
          <property name="minimumHeight"><number>60</number></property>
         </widget>
        </item>
       </layout>
      </widget>
      
      <!-- Page 2: Exam -->
      <widget class="QWidget" name="examPage">
       <layout class="QVBoxLayout" name="examLayout">
        <item>
         <layout class="QHBoxLayout" name="topLayout">
          <item>
           <widget class="QLabel" name="questionInfoLabel">
            <property name="text"><string>Question 1 of 10</string></property>
            <property name="styleSheet"><string>font-size: 16px; color: #a0a0a0;</string></property>
           </widget>
          </item>
          <item>
           <spacer name="spacer4">
            <property name="orientation"><enum>Qt::Horizontal</enum></property>
           </spacer>
          </item>
          <item>
           <widget class="QLabel" name="timerLabel">
            <property name="text"><string>30</string></property>
            <property name="styleSheet"><string>font-size: 36px; font-weight: bold; color: #27ae60;</string></property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QProgressBar" name="progressBar">
          <property name="value"><number>0</number></property>
          <property name="maximumHeight"><number>20</number></property>
         </widget>
        </item>
        <item>
         <widget class="QTextEdit" name="questionTextEdit">
          <property name="readOnly"><bool>true</bool></property>
          <property name="maximumHeight"><number>120</number></property>
          <property name="styleSheet"><string>background-color: #e3f2fd; color: #1565c0; font-size: 18px; font-weight: bold; padding: 15px; border: 2px solid #1976d2; border-radius: 8px;</string></property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="optionA">
          <property name="text"><string>Option A</string></property>
          <property name="styleSheet"><string>QRadioButton { font-size: 16px; padding: 10px; background-color: white; border: 2px solid #cccccc; border-radius: 8px; } QRadioButton::indicator { width: 20px; height: 20px; } QRadioButton:checked { background-color: #e3f2fd; border-color: #1976d2; }</string></property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="optionB">
          <property name="text"><string>Option B</string></property>
          <property name="styleSheet"><string>QRadioButton { font-size: 16px; padding: 10px; background-color: white; border: 2px solid #cccccc; border-radius: 8px; } QRadioButton::indicator { width: 20px; height: 20px; } QRadioButton:checked { background-color: #e3f2fd; border-color: #1976d2; }</string></property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="optionC">
          <property name="text"><string>Option C</string></property>
          <property name="styleSheet"><string>QRadioButton { font-size: 16px; padding: 10px; background-color: white; border: 2px solid #cccccc; border-radius: 8px; } QRadioButton::indicator { width: 20px; height: 20px; } QRadioButton:checked { background-color: #e3f2fd; border-color: #1976d2; }</string></property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="optionD">
          <property name="text"><string>Option D</string></property>
          <property name="styleSheet"><string>QRadioButton { font-size: 16px; padding: 10px; background-color: white; border: 2px solid #cccccc; border-radius: 8px; } QRadioButton::indicator { width: 20px; height: 20px; } QRadioButton:checked { background-color: #e3f2fd; border-color: #1976d2; }</string></property>
         </widget>
        </item>
        <item>
         <spacer name="spacer5">
          <property name="orientation"><enum>Qt::Vertical</enum></property>
         </spacer>
        </item>
        <item>
         <widget class="QPushButton" name="nextButton">
          <property name="text"><string>Next â†’</string></property>
          <property name="minimumHeight"><number>50</number></property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="submitButton">
          <property name="text"><string>Submit</string></property>
          <property name="minimumHeight"><number>50</number></property>
         </widget>
        </item>
       </layout>
      </widget>
      
      <!-- Page 3: Results -->
      <widget class="QWidget" name="resultsPage">
       <layout class="QVBoxLayout" name="resultsLayout">
        <item alignment="Qt::AlignHCenter">
         <widget class="QLabel" name="resultsTitle">
          <property name="text"><string>Results</string></property>
         </widget>
        </item>
        <item alignment="Qt::AlignHCenter">
         <widget class="QLabel" name="scoreLabel">
          <property name="text"><string>80%</string></property>
          <property name="styleSheet"><string>font-size: 72px; font-weight: bold; color: #27ae60;</string></property>
         </widget>
        </item>
        <item alignment="Qt::AlignHCenter">
         <widget class="QLabel" name="statusLabel">
          <property name="text"><string>PASS!</string></property>
          <property name="styleSheet"><string>font-size: 32px; font-weight: bold; color: #27ae60;</string></property>
         </widget>
        </item>
        <item alignment="Qt::AlignHCenter">
         <widget class="QLabel" name="detailsLabel">
          <property name="text"><string>Correct: 4 / 5</string></property>
          <property name="styleSheet"><string>font-size: 18px; color: #a0a0a0;</string></property>
         </widget>
        </item>
        <item alignment="Qt::AlignHCenter">
         <widget class="QLabel" name="timeUpLabel">
          <property name="text"><string/></property>
         </widget>
        </item>
        <item>
         <spacer name="spacer6">
          <property name="orientation"><enum>Qt::Vertical</enum></property>
         </spacer>
        </item>
        <item>
         <widget class="QPushButton" name="closeButton">
          <property name="text"><string>Close</string></property>
          <property name="minimumHeight"><number>50</number></property>
         </widget>
        </item>
       </layout>
      </widget>
      
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
